components/brave_new_tab_ui/containers/newTab/gridTile.tsx

```diff
@@ -3,158 +3,151 @@
   if (site.favicon === '') {
     return `chrome://favicon/size/64@1x/${site.url}`
   }
   return site.favicon
 }
 
-class TopSite extends React.PureComponent<Props, State> {
-  tileMenuRef: React.RefObject<any>
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      showMenu: false
-    }
-    this.tileMenuRef = React.createRef()
-  }
-
-  componentDidMount () {
-    document.addEventListener('mousedown', this.handleClickOutside)
-    document.addEventListener('mouseMove', this.handleMoveOutside)
-    document.addEventListener('keydown', this.onKeyPressSettings)
-  }
-
-  componentWillUnmount () {
-    document.removeEventListener('mousedown', this.handleClickOutside)
-    document.removeEventListener('mousemove', this.handleMoveOutside)
-    document.removeEventListener('keydown', this.onKeyPressSettings)
-  }
-
-  onKeyPressSettings = (event: KeyboardEvent) => {
-    if (event.key === 'Escape') {
-      this.setState({ showMenu: false })
-    }
-  }
-
-  handleClickOutside = (event: Event) => {
-    if (
-      this.tileMenuRef &&
-      this.tileMenuRef.current &&
-      !this.tileMenuRef.current.contains(event.target)
-    ) {
-      this.setState({ showMenu: false })
-    }
-  }
+export function SiteTile (props: { site: NewTab.Site, isMenuShowing?: boolean, children?: React.ReactNode, draggable?: ReturnType<typeof useSortable> }) {
+  const { site, isMenuShowing, children, draggable } = props
+  const style = useMemo(() => ({
+    transform: CSS.Transform.toString(draggable?.transform || null),
+    transition: draggable?.transition,
+    opacity: draggable?.isDragging ? 0 : 1
+  }), [draggable])
+
+  return <Tile
+    ref={draggable?.setNodeRef} {...draggable?.attributes} {...draggable?.listeners}
+    isDragging={!!draggable?.isDragging}
+    isMenuShowing={!!isMenuShowing}
+    title={site.title}
+    href={site.url}
+    style={style}>
+    {children}
+    <TileFavicon src={generateGridSiteFavicon(site)} />
+    <TileTitle>{site.title}</TileTitle>
+  </Tile>
+}
 
-  handleMoveOutside = (event: Event) => {
-    if (
-      this.tileMenuRef &&
-      this.tileMenuRef.current &&
-      !this.tileMenuRef.current.contains(event.target)
-    ) {
-      this.setState({ showMenu: false })
+function TopSite (props: Props) {
+  const { siteData, disabled } = props
+
+  const tileMenuRef = useRef<any>()
+  const sortable = useSortable({ id: siteData.id, disabled })
+  const [showMenu, setShowMenu] = useState(false)
+
+  const handleClickOutside = useCallback((e: Event) => {
+    if (!tileMenuRef.current || tileMenuRef.current.contains(e.target)) { return }
+    setShowMenu(false)
+  }, [])
+
+  const handleMoveOutside = useCallback((e: Event) => {
+    if (!tileMenuRef.current || tileMenuRef.current.contains(e.target)) { return }
+    setShowMenu(false)
+  }, [])
+
+  const onKeyPressSettings = useCallback((e: KeyboardEvent) => {
+    if (e.key === 'Escape') { setShowMenu(false) }
+  }, [])
```

Comment:
nit: since these are only used in 1 place - inside the `useEffect`, might be cleaner to just define the functions (without `useCallback`) inside there and not have the useEffect depend on them (since they won't change ever, anyway)?

Reply:
Done. I also noticed that the code which adds the `mousemove` handler is attaching to `mouseMove` (it was like this before the change). I've removed the handler completely because I don't think hiding the menu on mousemove is desired behavior.